### ISSUE TITLE
Make timelineVariable(..) available in conditional functions

### ISSUE THREAD

**Issue by wolfgangwalther** [jspsych/jsPsych#574]
I would like to do this:

```    
{
  timeline_variables: XYZ,
  timeline: [
    {
      conditional_function: function() {
        return jsPsych.timelineVariable('X', true)
      },
      timeline: [
        {
          ....
        }
      ]
    },
    ...
  ]
}
```

But I'm getting:

```
TypeError: timeline_parameters.timeline[progress.current_location] is undefined (jspsych.js:566:9)
  TimelineNode/this.timelineVariable (jspsych.js:566:9)
  TimelineNode/this.timelineVariable (jspsych.js:566:16)
  TimelineNode/this.timelineVariable (jspsych.js:566:16)
  core.timelineVariable (jspsych.js:296:14)
  conditional_function (experiment.js:[...]:[...])
[...]
```

I would expect jsPsych.timelineVariable to work here.

**Comment by jodeleeuw** [comment]
Currently the `conditional_function` runs *prior* to the first trial on the timeline, so technically there is no active trial from which to draw the timeline variables. In your use case, are you wanting the `timelineVariable()` function to return the value of the variable for the first trial on that timeline?

**Comment by wolfgangwalther** [comment]
But the conditional_function is prior to the first trial of the 2nd (nested) timeline, correct? That means we are actually running the outer timeline already. When the outer timeline is run - shouldn't there already be some timelineVariables associated with it?
I understand your point if we were talking about timeline_variables set on the inner trial, but this is not the case here.

Maybe to illustrate my point better:
```
{
  timeline_variables: XYZ,
  timeline: [
    {
      type: 'something',
      stimulus: jsPsych.timelineVariable('Y'),
      ...
    },
    {
      conditional_function: function() {
        return jsPsych.timelineVariable('X', true)
      },
      timeline: [
        {
          ....
        }
      ]
    },
    ...
  ]
}
```

Clearly the trial 'something' is run *before* evaluating the conditional_function, is it?

That means we should already have timeline variables set up - but this will still throw the same error.

**Comment by wolfgangwalther** [comment]
And to answer your actual question, now that I have hopefully explained better:
I want `timelineVariable()` to return the value of the variable of the *current* trial of the **outer** timeline.

Does that make sense?

**Comment by jodeleeuw** [comment]
I missed the fact that these were nested. This makes sense, and I agree
with you that it should work.

**Comment by jodeleeuw** [comment]
This is fixed and will be released in 6.1
